src/components/Calendar/YearPicker.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useEffect, useRef } from 'react'
+import styled, { css } from 'styled-components'
+
+import { Theme, useTheme } from '../../hooks/useTheme'
+
+type Props = {
+  selectedYear?: number
+  fromYear: number
+  toYear: number
+  onSelectYear: (year: number) => void
+}
+
+export const YearPicker: FC<Props> = ({ selectedYear, fromYear, toYear, onSelectYear }) => {
+  const themes = useTheme()
+  const scrollingRef = useRef<HTMLDivElement>(null)
+
+  const thisYear = new Date().getFullYear()
+  const numOfYear = Math.max(Math.min(toYear, 9999) - fromYear + 1, 0)
+  const yearArray = Array(numOfYear)
+    .fill(null)
+    .map((_, i) => fromYear + i)
+
+  useEffect(() => {
+    if (scrollingRef.current) {
+      scrollingRef.current.scrollIntoView({ block: 'center' })
+    }
+  }, [])
+
+  return (
+    <Container>
+      <Grid>
+        {yearArray.map((year) => {
+          const isThisYear = thisYear === year
+          return (
+            <Cell key={year} themes={themes} onClick={() => onSelectYear(year)}>
+              <YearWrapper
+                themes={themes}
+                isThisYear={isThisYear}
+                isSelected={selectedYear === year}
+                ref={isThisYear ? scrollingRef : null}
+              >
+                {year}
+              </YearWrapper>
+            </Cell>
+          )
+        })}
+      </Grid>
+    </Container>
+  )
+}
+
+const Container = styled.div`
+  width: 100%;
+  height: 100%;
+  padding: 8px 2.5px;
```

Comment:
Thanks!
In the design file,
* margin between cells: `27px`
* margin between cell and side edge: `16px`

so I calcurated horizontal padding of Container by `16 - (27 / 2) = 2.5`.
But it is more important to avoid using subpixel, I change it to `3px`.